post_request_task/tests.py

```diff
@@ -89,4 +124,33 @@ def test_deduplication(self):
         test_task.delay()
         test_task.delay()
 
-        self._verify_task_filled()
+        self._verify_one_task_queued()
+
+    def test_tasks_deduplication_with_different_arguments(self):
+        """Test arguments sent to the task are used when de-duping."""
+        request_started.send(sender=self)
+        test_task_with_args_and_kwargs.delay(42)
+        test_task_with_args_and_kwargs.delay(42, bar='rista')
+        test_task_with_args_and_kwargs.delay(42, bar='rista')
+
+        self._verify_x_tasks_queued(
+            2,  # 2 tasks should have been kept.
+            ((42,), (42,)),  # Each of those tasks sent "42" as the only arg...
+            ({}, {'bar': 'rista'}),  # ... Only the second one sent kwargs.
+            test_task=test_task_with_args_and_kwargs)
+
+    def test_tasks_deduplication_with_different_keyword_arguments(self):
+        """Test keyword arguments sent to the task are used when de-duping."""
+        request_started.send(sender=self)
+        test_task_with_args_and_kwargs.delay(42, bar='rista')
+        test_task_with_args_and_kwargs.delay(42, bar='rista')
+        test_task_with_args_and_kwargs.delay(42, bar='atheon')
+        test_task_with_args_and_kwargs.delay(42, bar='rage')
+        test_task_with_args_and_kwargs.delay(42, bar='rista')
+        test_task_with_args_and_kwargs.delay(42, bar='rage')
+
+        self._verify_x_tasks_queued(
+            3,
+            ((42,), (42,), (42, )),
+            ({'bar': 'rista'}, {'bar': 'atheon'}, {'bar': 'rage'}),
+            test_task=test_task_with_args_and_kwargs)
```

Comment:
How should deduplication work wrt `extrakw`? There's no test for it but it's checking the whole tuple so it would compare that as well. Can we added tests for `extrakw` here and above similar to `test_task_with_args_and_kwargs`?